110 Incompatible Modules

shanefjordan opened the pull request:
This is to resolve the Drutiny-cs-adaptor issue #110.

briantully commented:
Awesome work, Shane! These new policies will be extremely helpful :)

In terms of process, do we need to squash recaptcha (79733dc & 2362c85) and varnish (652b2c5 & fdc8164)?

fiasco commented on Policy/drupal-7/apcDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Best Practice
+  - Performance
+description: |
+  Alternative PHP cache is not recommended on Acquia Cloud. It stores 
```
Unless this is an Acquia specific policy, it shouldn't have references to Acquia Cloud as its an open source policy.

fiasco commented on Policy/drupal-7/autoSlaveDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Best Practice
+  - Performance
+description: |
+  The Acquia Cloud platform is configured to auto-detect the primary and 
```
Another reference to Acquia Cloud

fiasco commented on Policy/drupal-7/civiCRMDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Performance
+description: |
+  CiviCRM is dependent on stored procedures and triggers that do not work 
+  on the Acquia Cloud platform. It is   recommended to disable and 
```
Another reference to Acquia Cloud

fiasco commented on Policy/drupal-7/configMgmtDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Performance
+description: |
+  This module requires a writeable directory that is also trackable at the 
+  same time by Git; this is not currently possible on Acquia Cloud. It is 
```
Another reference to Acquia Cloud

fiasco commented on Policy/drupal-7/fbconnect.policy.yml:
```diff
@@ -0,0 +1,19 @@
+title: Facebook Connect is not installed
+name: Drupal-7:fbconnectDisabled
```
see https://drutiny.github.io/2.3.x/policy/#depends

You could configure this policy to omit itself if the module isn't enabled:

```
depends:
  - expression: drupal_module_enabled('fbconnect')
    on_fail: omit
```

This way it reduces the "noise" of contextual policies.

fiasco commented on Policy/drupal-7/fileCacheDisabled.policy.yml:
```diff
@@ -0,0 +1,21 @@
+title: File Cache is not installed
+name: Drupal-7:fileCacheDisabled
```
Since this policy is gluster specific it should have Gluster in the name.

fiasco commented on Policy/drupal-7/memcacheStorageDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Best Practice
+  - Performance
+description: |
+  Although not incompatible, Acquia discourages using this module’s due 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-7/purgeDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+description: |
+  The Purge (7.x-1.x) module is not specifically incompatible, but can be 
+  difficult to set up correctly. We suggest using Acquia Purge instead. 
+  It is specifically intended for use on Acquia Cloud. Over time, these 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-7/serialDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Best Practice
+  - Performance
+description: |
+  Serial is based on an auto_increment of 1. Acquia Cloud uses an 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-7/shibAuthDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Best Practice
+  - Performance
+description: |
+  Shibboleth is not supported on Acquia Cloud. Other methods of achieving this 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-7/varnishModuleDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Performance
+description: |
+  This Drupal module attempts to replicate the effort of the Varnish Cache 
+  that is already available to Acquia Cloud applications. It will not work 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/apacheSolrFileDisabled.policy.yml:
```diff
@@ -0,0 +1,19 @@
+  - Best Practice
+  - Performance
+description: |
+  Customers are unable to modify the solrconfig.xml file on Acquia Cloud. 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/apcDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Best Practice
+  - Performance
+description: |
+  Alternative PHP cache is not recommended on Acquia Cloud. It stores 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/memcacheStorageDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+  - Best Practice
+  - Performance
+description: |
+  Although not incompatible, Acquia discourages using this module’s due 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/serialDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Best Practice
+  - Performance
+description: |
+  Serial is based on an auto_increment of 1. Acquia Cloud uses an 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/shibAuthDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+  - Best Practice
+  - Performance
+description: |
+  Shibboleth is not supported on Acquia Cloud. Other methods of achieving this 
```
Reference to Acquia. Should perhaps be an Acquia specific policy?

fiasco commented on Policy/drupal-8/smartIPDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+title: Smart IP is not installed
+name: Drupal-8:smartIPDisabled
+class: \Drutiny\Plugin\Drupal8\Audit\ModuleDisabled
```
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

fiasco commented on Policy/drupal-8/supercookieDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+title: Super Cookie is not installed
+name: Drupal-8:supercookieDisabled
+class: \Drutiny\Plugin\Drupal8\Audit\ModuleDisabled
```
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

fiasco commented on Policy/drupal-8/tbMegaMenuDisabled.policy.yml:
```diff
@@ -0,0 +1,21 @@
+title: TB Mega Menu is not installed
+name: Drupal-8:tbMegaMenuDisabled
```
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

fiasco commented on Policy/drupal-8/textSizeDisabled.policy.yml:
```diff
@@ -0,0 +1,20 @@
+title: Text Size is not installed
+name: Drupal-8:textSizeDisabled
```
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

fiasco commented on Policy/drupal-8/varnishModuleDisabled.policy.yml:
```diff
@@ -0,0 +1,22 @@
+description: |
+  This Drupal module attempts to replicate the effort of the Varnish Cache 
+  that is already available to Acquia Cloud applications. It will not work 
+  with Acquia Cloud applications because it requires connections to the 
```
Acquia specific policy?
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

fiasco commented on Policy/drupal-8/viewsFilterHarmonizerDisabled.policy.yml:
```diff
@@ -0,0 +1,19 @@
+title: Views Filter Harmonizer is not installed
+name: Drupal-8:viewsFilterHarmonizerDisabled
```
another policy that could be omitted if the module doesn't exist: https://drutiny.github.io/2.3.x/policy/#depends

shanefjordan commented:
I've cleaned up the policies that would be more dependent on Acquia hosting. I've also added the depend statements to the policies to keep reporting clean.

shanefjordan commented:
@fiasco The checks that are related to Acquia Cloud, should I move them to the cs-adaptor project? If so, I believe I'll have to separate them into subfolders for drupal-8 and drupal-7 since the names are the same. 

I wanted to check before making the changes and having to move everything again.

fiasco commented:
No, there are acquia policies in this repository, they just need to have Acquia in the name of them since they are realm specific.

I think there is also an Acquia folder that they should be placed into. That folder can have drupal7/8 subfolders too.

shanefjordan commented:
@fiasco Everything should be sorted into the proper locations now and ready for review.

fiasco commented:
Thanks Shane, this is awesome!
